lightning/src/chain/channelmonitor.rs

```diff
@@ -999,54 +1023,255 @@ impl<Signer: Sign> ChannelMonitor<Signer> {
-			pending_events: Vec::new(),
+				payment_preimages: HashMap::new(),
+				pending_monitor_events: Vec::new(),
+				pending_events: Vec::new(),
 
-			onchain_events_waiting_threshold_conf: HashMap::new(),
-			outputs_to_watch,
+				onchain_events_waiting_threshold_conf: HashMap::new(),
+				outputs_to_watch,
 
-			onchain_tx_handler,
+				onchain_tx_handler,
 
-			lockdown_from_offchain: false,
-			holder_tx_signed: false,
+				lockdown_from_offchain: false,
+				holder_tx_signed: false,
 
-			last_block_hash: Default::default(),
-			secp_ctx,
+				last_block_hash: Default::default(),
+				secp_ctx,
+			}),
 		}
 	}
 
+	#[cfg(test)]
+	fn provide_secret(&self, idx: u64, secret: [u8; 32]) -> Result<(), MonitorUpdateError> {
+		self.inner.lock().unwrap().provide_secret(idx, secret)
+	}
+
+	/// Informs this monitor of the latest counterparty (ie non-broadcastable) commitment transaction.
+	/// The monitor watches for it to be broadcasted and then uses the HTLC information (and
+	/// possibly future revocation/preimage information) to claim outputs where possible.
+	/// We cache also the mapping hash:commitment number to lighten pruning of old preimages by watchtowers.
+	pub(crate) fn provide_latest_counterparty_commitment_tx<L: Deref>(
+		&self,
+		txid: Txid,
+		htlc_outputs: Vec<(HTLCOutputInCommitment, Option<Box<HTLCSource>>)>,
+		commitment_number: u64,
+		their_revocation_point: PublicKey,
+		logger: &L,
+	) where L::Target: Logger {
+		self.inner.lock().unwrap().provide_latest_counterparty_commitment_tx(
+			txid, htlc_outputs, commitment_number, their_revocation_point, logger)
+	}
+
+	#[cfg(test)]
+	fn provide_latest_holder_commitment_tx(
+		&self,
+		holder_commitment_tx: HolderCommitmentTransaction,
+		htlc_outputs: Vec<(HTLCOutputInCommitment, Option<Signature>, Option<HTLCSource>)>,
+	) -> Result<(), MonitorUpdateError> {
+		self.inner.lock().unwrap().provide_latest_holder_commitment_tx(
+			holder_commitment_tx, htlc_outputs)
+	}
+
+	#[cfg(test)]
+	pub(crate) fn provide_payment_preimage<B: Deref, F: Deref, L: Deref>(
+		&self,
+		payment_hash: &PaymentHash,
+		payment_preimage: &PaymentPreimage,
+		broadcaster: &B,
+		fee_estimator: &F,
+		logger: &L,
+	) where
+		B::Target: BroadcasterInterface,
+		F::Target: FeeEstimator,
+		L::Target: Logger,
+	{
+		self.inner.lock().unwrap().provide_payment_preimage(
+			payment_hash, payment_preimage, broadcaster, fee_estimator, logger)
+	}
+
+	pub(crate) fn broadcast_latest_holder_commitment_txn<B: Deref, L: Deref>(
+		&self,
+		broadcaster: &B,
+		logger: &L,
+	) where
+		B::Target: BroadcasterInterface,
+		L::Target: Logger,
+	{
+		self.inner.lock().unwrap().broadcast_latest_holder_commitment_txn(broadcaster, logger)
+	}
+
+	/// Updates a ChannelMonitor on the basis of some new information provided by the Channel
+	/// itself.
+	///
+	/// panics if the given update is not the next update by update_id.
+	pub fn update_monitor<B: Deref, F: Deref, L: Deref>(
+		&self,
```

Comment:
nit: I find rustfmt-style unwrapping like this rather hard to read, its something that my head thinks shouldn't be vertical but is (and I have to check it against the use which is horizontal), and suddenly I can't see as much context.

Reply:
I was resistant to this for awhile until dealing with extremely long signatures in `lightning-block-sync`. I think using this format for long signatures is a good convention because it makes the parameters easy to scan and provides a consistent rule for formatting.

Otherwise, parameters can be easily missed if they are bin packed. It may seem a little odd at first, but it doesn't take long to adjust to reading this format. And it has the advantage of clearly indicating a parameter's type. This otherwise becomes especially difficult for more elaborate types like tuples or those with more than one type parameter, as the additional commas make the parameter list harder to scan.

That said, I do prefer to bin pack arguments at the call sites as I've done throughout in cases where the line exceeds the 100-character limit. Here, there are no types so any advantage of breaking each argument across lines is lost. I think this is a good middle ground as the additional vertical spacing is limited to signatures.

Reply:
It is somewhere in between, indeed, and I see the value of it, but I'd argue that more things are missed in review due to missing context than scan errors. Sometimes just a few lines is needed, but sometimes even the next function up is important (or being able to fit the full parameter list while looking at a function). Sadly, different people have different terminal heights so there's not an easy fix.